lms/djangoapps/courseware/model_data.py

```diff
@@ -275,28 +275,26 @@ def find_or_create(self, key):
             return field_object
 
         if key.scope == Scope.user_state:
-            field_object, __ = StudentModule.objects.get_or_create(
+            field_object = StudentModule(
                 course_id=self.course_id,
                 student_id=key.user_id,
                 module_state_key=key.block_scope_id,
-                defaults={
-                    'state': json.dumps({}),
-                    'module_type': key.block_scope_id.block_type,
-                },
+                state=json.dumps({}),
+                module_type=key.block_scope_id.block_type,
             )
```

Comment:
No. If we make it past line 274, then there is no StudentModule (at least, not one in the context of your transaction), so you can immediately create one. (Unless you didn't tell FDC to prefetch the right things, but in that case, all of your reads are based on the wrong data anyway, so I'd rather the writes fail).